components/locid_transform/src/expander.rs

```diff
@@ -477,83 +515,165 @@ impl LocaleExpander {
     /// assert_eq!(lc.minimize(&mut locale), TransformResult::Unmodified);
     /// assert_eq!(locale, locale!("zh"));
     /// ```
-    pub fn minimize<T: AsMut<LanguageIdentifier>>(&self, mut langid: T) -> TransformResult {
+    pub fn minimize<T: AsMut<LanguageIdentifier>>(&self, langid: T) -> TransformResult {
+        self.minimize_with_options(langid, LocaleExpanderOptionsBag::default())
+    }
+
+    /// This returns a new Locale that is the result of running the
+    /// 'Remove Likely Subtags' algorithm from
+    /// <https://www.unicode.org/reports/tr35/#Likely_Subtags>
+    /// with the options given by [`LocaleExpanderOptionsBag`].
+    ///
+    /// If the result of running the algorithm would result in a new locale, the
+    /// locale argument is updated in place to match the result, and the method
+    /// returns [`TransformResult::Modified`]. Otherwise, the method
+    /// returns [`TransformResult::Unmodified`] and the locale argument is
+    /// unchanged.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::locale;
+    /// use icu_locid_transform::{
+    ///     LocaleExpander, LocaleExpanderOptionsBag, LocaleExpanderSubtagPriority,
+    ///     TransformResult
+    /// };
+    ///
+    /// let lc = LocaleExpander::new();
+    ///
+    /// let mut locale = locale!("zh_TW");
+    /// assert_eq!(
+    ///     lc.minimize_with_options(
+    ///         &mut locale,
+    ///         LocaleExpanderOptionsBag::from_subtag_priority(
+    ///             LocaleExpanderSubtagPriority::Script
+    ///         )
+    ///     ),
+    ///     TransformResult::Modified
+    /// );
+    /// assert_eq!(locale, locale!("zh_Hant"));
+    /// ```
+    pub fn minimize_with_options<T: AsMut<LanguageIdentifier>>(
```

Comment:
Suggestion: Looking at this now, I think we should just have two functions `minimize` and `minimize_favor_script` and skip the options bag thing.